submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,100 @@
+//game score
+var score = 0
+//Enemy class with update(), render() methods
+var Enemy = function (x, y, speed) {
+    this.x = x
+    this.y = y
+    this.speed = speed
+    this.sprite = 'images/car.png';
+};
+var END_CANVAS = 510
+var BEGIN_CANVAS = -50
+Enemy.prototype.update = function (dt) {
+
+    this.x += this.speed * dt
+    if (this.x > END_CANVAS) {
+        this.x = BEGIN_CANVAS
+        //random speed for enemies
+        this.speed = 100 + Math.floor(Math.random() * 222)
+    }
+    // return to start position player when collide with enemy
+    if (player.x > (this.x - 60) && player.x < (this.x + 60) && player.y > (this.y - 60) && player.y < (this.y + 60)) {
+        player.startPosition();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var allEnemies = []
+var enemyLocation = [60, 140, 220]
+
+enemyLocation.forEach(function (y) {
+    allEnemies.push(new Enemy(0, y, 200))
+})
```

Comment:
This is a use case for [`Array#map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

Basically you want to convert an array of y-coordinates into an array of initialized objects. Try to refactor this part.